### DISCUSSION TITLE
adding an unsupported device that has unprintable characters in the model number

### DISCUSSION THREAD

**Discussion by dreed47** [Koenkk/zigbee2mqtt#15036] [Devices]
I purchased a few Ecosmart D1533 bulbs that I thought were supported but are not.  So I proceeded to follow the process for adding a new unsupported device.  The logs produced this message 

`Device '0xdd16abfff1a1a0aa' with Zigbee model ' T Z ' and manufacturer name '2.1.0_16d_YF05' is NOT supported`

Only the model is not really "space T space Z space" as it looks like it has some non-printable characters in it.  When I look at the code in ecosmart.js under the D1533 device I see an array of zigbeeModels that look like 

`'\u0000\u0002\u0000\u0004\u0000\f]�\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u000e'`

First issue is how do I decipher what this device is producing for model info and the second issue is how do I add a MODEL.js file to my local install to be able to test it out when this model has non-printable characters?

Message #hardware

**Comment by dreed47** [comment]
I found the model string from the database.db file and added a local D1533.js file and tested it locally.  Pull request created to add new model to ecosmart.js  

https://github.com/Koenkk/zigbee-herdsman-converters/pull/4969

**Comment by mominsaggu** [comment]
Hi, i have the same problem, also, i am new to all this, can you please guide me on where to find the Database.db? i am running HA on Virtualbox. thanks!

**Reply by andrew-wylie** [reply]
> Hi, i have the same problem, also, i am new to all this, can you please guide me on where to find the Database.db? i am running HA on Virtualbox. thanks!

Hi, the way I got the ID, is go to Zigbee2mqtt ... Settings ... Tools ... Download State, and open the json file within. Then I looked for the ieeeAddr value and found the modelID nearby. I copied and replaced \\ with \ and dropped that into my js file, restarted, and now it sees it.
